numpy/_core/src/multiarray/alloc.c

```diff
@@ -30,13 +30,15 @@
 /* Do not enable the alloc cache if the GIL is disabled, or if ASAN or MSAN
  * instrumentation is enabled. The cache makes ASAN use-after-free or MSAN
  * use-of-uninitialized-memory warnings less useful. */
-#define USE_ALLOC_CACHE 1
 #ifdef Py_GIL_DISABLED
-# define USE_ALLOC_CACHE 0
+  #define USE_ALLOC_CACHE 0
 #elif defined(__has_feature)
-# if __has_feature(address_sanitizer) || __has_feature(memory_sanitizer)
-#  define USE_ALLOC_CACHE 0
-# endif
+  #if __has_feature(address_sanitizer) || __has_feature(memory_sanitizer)
+    #define USE_ALLOC_CACHE 0
+  #endif
+#endif
+#ifndef USE_ALLOC_CACHE
+  #define USE_ALLOC_CACHE 1
```

Comment:
Ah, thanks for fixing this warning!  Is this already in main?

Reply:
PR incoming.